schema/artifact_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/opencontainers/image-spec/schema"
+)
+
+func TestArtifact(t *testing.T) {
+	for i, tt := range []struct {
+		manifest string
+		fail     bool
+	}{
+		// expected failure: mediaType does not match pattern
+		{
+			manifest: `
+{
+  "mediaType" : "invalid",
+  "artifactType" : "application/example",
+  "blobs": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": 148,
+      "digest": "sha256:c57089565e894899735d458f0fd4bb17a0f1e0df8d72da392b85c9b35ee777cd"
+    }
+  ]
+}
+`,
+			fail: true,
+		},
+
+		// expected failure: invalid artifact mediaType
+		{
+			manifest: `
+{
+  "mediaType" : "application/vnd.oci.artifact.manifest.v1+json",
+  "artifactType" : "invalid",
+  "blobs": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": "148",
+      "digest": "sha256:c57089565e894899735d458f0fd4bb17a0f1e0df8d72da392b85c9b35ee777cd"
+    }
+  ]
+}
+`,
+			fail: true,
+		},
+
+		// expected failure: blob[0].size is a string, expected integer
+		{
+			manifest: `
+{
+  "mediaType" : "application/vnd.oci.artifact.manifest.v1+json",
+  "artifactType" : "application/example",
+  "blobs": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": "148",
+      "digest": "sha256:c57089565e894899735d458f0fd4bb17a0f1e0df8d72da392b85c9b35ee777cd"
+    }
+  ]
+}
+`,
+			fail: true,
+		},
+
+		// expected failure: subject: size is required
+		{
+			manifest: `
+{
+  "mediaType" : "application/vnd.oci.artifact.manifest.v1+json",
+  "artifactType" : "application/example",
+ "subject": {
+    "mediaType": "application/vnd.oci.image.manifest.v1+json",
+    "digest": "sha256:c86f7763873b6c0aae22d963bab59b4f5debbed6685761b5951584f6efb0633b"
+  },
+  "annotations": {
+    "key1": "value1",
+    "key2": "value2"
+  }
```

Comment:
Nit, indentation needs an extra space

Reply:
I went ahead and reformatted all the examples. It seems that certain properties had spaces after the property names and some didn't.

Reply:
Please feel to add more examples. I'm hoping to get the baseline framework and schema in first.

Reply:
@sudo-bmitch hope I've addressed the issues.